CodeEdit/Documents/WorkspaceDocument.swift

```diff
@@ -267,37 +267,25 @@ import TabBar
                                                object: nil)
     }
 
+    private func readSelectionState() throws -> WorkspaceSelectionState {
+        guard let path = fileURL?.path,
+              let pathData = path.data(using: .utf8) else { return selectionState }
+        let hash = String(CryptoKit.SHA256.hash(data: pathData).hashValue)
```

Comment:
`.hashValue` will not be equal across executions of the program ([source](https://developer.apple.com/documentation/swift/string/hashvalue)).

Consider implementing it something like that:

```swift
let stringData = "Hello World".data(using: .utf8)!

let hash = SHA256.hash(data: stringData)

let hashString = hash.compactMap { String(format: "%02x", $0) }.joined()
```

Reply:
Maybe consider creating an Extension on `String` like so:

```swift
extension String {
    var sha256Hash: String? {
        guard let data = self.data(using: .utf8) else { return nil }
        let hash = SHA256.hash(data: data)
        return hash.compactMap { String(format: "%02x", $0) }.joined()
    }
}

// call like so:

"Hello World".sha256Hash
```